docs/enhancements/installation-progress-bar.md

```diff
@@ -0,0 +1,113 @@
+# Cluster installation progress bar
+
+## Goals
+
+* Progress should be computed in the BE by design and not in the UI
+  * that will also make it usable from other APIs as kube-api
+* Installed clusters with failed non-essential workers or OLM-operators should still reach 100% progress.
+
+
+## Current implementation
+
+* Computed in UI
+* `progressPercent = hostsProgressPercent * 0.75 + operatorsProgressPercent * 0.25`
+  * `operatorsProgressPercent =(completedOperators / monitoredOperators) * 100`
+    * A completedOperator is one of the following:
+      * A built-in operator with `available` status
+      * An OLM operator with `available` or `failed` status
+  * `hostsProgressPercent = (ΣcompletedStages / ΣallStages) * 100`
+    * For each host the data is retrieved by using the following APIs
+
+```
+GET api/assisted-install/v1/clusters/{cluster_id} | jq '.hosts[0].progress'
+{
+  "current_stage": "Done",
+  "stage_started_at": "2021-05-31T13:37:26.224Z",
+  "stage_updated_at": "2021-05-31T13:37:26.224Z"
+}
+
+GET /api/assisted-install/v1/clusters/{cluster_id} | jq '.hosts[0].progress_stages'
+[
+  "Starting installation",
+  "Installing",
+  "Writing image to disk",
+  "Waiting for control plane",
+  "Rebooting",
+  "Configuring",
+  "Joined",
+  "Done"
+]
+```
+
+## New implementation
+
+* Will be computed in BE
+* `progres =ΣclusterStageProgress * Wi = preparingForInstallation * Wpi + installing * Wi + finalizing * Wf` while `ΣWi=1`
+  * We can manually compute an avg duration for the previous installations based on events to choose those weights and hardcode them.
+
+
+#### Preparing for installation
+
+This stage mostly generates manifests and installation configs, therefore, we will set  `preparingForInstallationProgress={1, 0}` depending on if it is done or not.
+
+
+#### Installing
+
+* This stage refers to hosts' installation
+* We will “assume” all host’s stages have a similar duration, therfore, they will be equally weighted.  
+* Hosts' progresses will be computed the same way we it is done in the current implementation, aka `hostsProgressPercent = ΣcompletedStages / ΣallStages`
+
+
+#### Finalizing
+
+* This stage refers to operators' installation
+* We will “assume” all operatos have a similar installation duration.
+* Operators' progresses will be computed the same way we it is done in the current implementation, aka `operatorsProgressPercent = completedOperators / monitoredOperators`
+
+Suggested APIs:
+
+```
+GET /api/assisted-install/v1/clusters/{cluster_id}
+{
+  "id": "3a745598-dde4-46dd-bdb1-ce7e1dea119b",
+  "progress":                                    <-------- NEW
```

Comment:
maybe we should call it installation_progress

Reply:
It is called `progress` in any other APIs (you can see it in the suggested API which is part of this PR).